shell/platform/fuchsia/flutter/component_v2.cc

```diff
@@ -13,7 +13,6 @@
 #include <lib/fdio/io.h>
 #include <lib/fdio/namespace.h>
 #include <lib/ui/scenic/cpp/view_ref_pair.h>
```

Comment:
Ideally we could remove the usage of "ViewRefPair" - it's just a simple std::pair<ViewRef, ViewRefControl> and a "Create" utility function.  This allows to break the dependency on "lib/ui/scenic" entirely.

Is Scenic team planning to keep that entire SDK library around just for the "ViewRefPair" construct - seems like a unnecesary maintenance burden?

I'd be quite happy if you broke the dependency here.  Otherwise, please file a bug.

Reply:
Is your recommendation to copy the `ViewRefPair` struct to the `shell/platform/fuchsia/flutter` directory and use that instead of the sdk dependency on it? Otherwise, I attempted to migrate to use a `std::pair<ViewRef, ViewRefControl>` but got this compilation error:
```
/usr/local/google/home/carolineliu/engine/src/buildtools/linux-x64/clang/bin/../include/c++/v1/__memory/unique_ptr.h:689:34: error: call to implicitly-deleted copy constructor of 'std::pair<fuchsia::ui::views::ViewRefControl, fuchsia::ui::views::ViewRef>'
  689 |   return unique_ptr<_Tp>(new _Tp(_VSTD::forward<_Args>(__args)...));
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/google/home/carolineliu/engine/src/buildtools/linux-x64/clang/bin/../include/c++/v1/__config:805:17: note: expanded from macro '_VSTD'
  805 | #  define _VSTD std
      |                 ^
../../flutter/shell/platform/fuchsia/flutter/component_v2.cc:649:31: note: in instantiation of function template specialization 'std::make_unique<flutter_runner::Engine, flutter_runner::ComponentV2 &, const std::string &, std::shared_ptr<sys::ServiceDirectory> &, std::shared_ptr<sys::ServiceDirectory> &, flutter::Settings &, fuchsia::ui::views::ViewCreationToken, std::pair<fuchsia::ui::views::ViewRefControl, fuchsia::ui::views::ViewRef> &, fml::UniqueObject<fdio_namespace *, flutter_runner::UniqueFDIONSTraits>, fidl::InterfaceRequest<fuchsia::io::Directory>, flutter_runner::FlutterRunnerProductConfiguration &, std::vector<std::string>, bool>' requested here
  649 |   shell_holders_.emplace(std::make_unique<Engine>(
      |                               ^
/usr/local/google/home/carolineliu/engine/src/buildtools/linux-x64/clang/bin/../include/c++/v1/__utility/pair.h:92:27: note: explicitly defaulted function was implicitly deleted here
   92 |     _LIBCPP_HIDE_FROM_ABI pair(pair const&) = default;
      |                           ^
/usr/local/google/home/carolineliu/engine/src/buildtools/linux-x64/clang/bin/../include/c++/v1/__utility/pair.h:89:9: note: copy constructor of 'pair<fuchsia::ui::views::ViewRefControl, fuchsia::ui::views::ViewRef>' is implicitly deleted because field 'first' has a deleted copy constructor
   89 |     _T1 first;
      |         ^
gen/build/fuchsia/fidl/fuchsia/ui/views/cpp/fidl.h:875:19: note: copy constructor of 'ViewRefControl' is implicitly deleted because field 'reference' has a deleted copy constructor
  875 |   ::zx::eventpair reference;
      |                   ^
../../fuchsia/sdk/linux/pkg/zx/include/lib/zx/eventpair.h:24:3: note: copy constructor is implicitly deleted because 'eventpair' has a user-declared move constructor
   24 |   eventpair(eventpair&& other) : object(other.release()) {}
      |   ^
../../flutter/shell/platform/fuchsia/flutter/engine.h:58:49: note: passing argument to parameter 'view_ref_pair' here
   58 |                    fuchsia::ui::views::ViewRef> view_ref_pair,
      |                                                 ^
```

In the mean time, I'll file a bug to unblock this PR on this conversation.

Reply:
Chatted about this offline, this should be fixed now. :) Thanks for the help!